### ISSUE TITLE
extra column when passing in comma ',' or \r\n

### ISSUE THREAD

**Issue by liuhe2020** [dolezel/react-csv-downloader#373]
I have an string for address like this '5 Example Street, Example Town, Example Country', the output csv will produce 3 seperate columns instead of one. The same goes if the string is '5 Example Street\r\nExample Town\r\nExample Country'.

In the end I resolved to using react-csv...

**Comment by dolezel** [comment]
You need to change the separator or wrapColumnChar

**Comment by jwdotjs** [comment]
For anyone else who runs into this problem, I tried using `wrapColumnChar="'"`, but it made my CSV values (including) headers an unreadable mess, and I needed the separator to stay as `,` for consistency when uploading and downloading CSVs so neither of @dolezel's suggestions really worked for me.

I ended up using the following solution:

```javascript
<CSVDownloader
  columns={Object.keys(records?.[0] ?? {}).map((key) => ({
    id: key,
    displayName: key,
  }))}
  datas={() =>
    records.reduce((acc, record) => {
      const newRecord: Record<string, any> = {};

      Object.keys(record).forEach((key) => {
        let value = record[key];

        value = Array.isArray(value) ? value.join(',') : value;

        value =
          typeof value === 'string' &&
          (value.includes(',') || value.includes('\n')) &&
          !value.includes('"')
            ? `"${value}"`
            : value;

        if (value !== null && value !== undefined) {
          newRecord[key] = String(value);
        } else {
          newRecord[key] = null;
        }
      });

      acc.push(newRecord);

      return acc;
    }, [] as Record<string, any>[])
  }
  extension=".csv"
  separator=","
  filename={`incentives-${new Date().toISOString()}`}
  handleError={(error) => {
    console.error(error);
  }}
>
  <Button variant="contained" startIcon={<DownloadIcon />}>
    Download CSV
  </Button>
</CSVDownloader>
```

Essentially, when `Download CSV` is clicked, it performs a reduce on the data, checks if the value is a string that has `,`, or `\n` characters and is not already wrapped in `"`. If those criteria are met, it wraps that string in `"` so for example `first, second, third` would be transformed to `"first, second, third"`. 

This solution works for me and hopefully helps others that run into the same problem. 

I'm not sure if there are any negative tradeoffs to this approach. Arguable, this library automatically escape or wrap CSV-breaking characters within strings, but if that doesn't happen or isn't in the vision, this at least should resolve the primary issue that the OP and I ran into. (If @dolezel likes this approach, I'm happy to throw up a PR and take lead on that)

Cheers all.

**Comment by Gijuno** [comment]
TLDR;
Just wrap the data with `"`
It goes on.

ex) `first, second, third` to `"first, second, third"`
